gnpy/core/elements.py

```diff
@@ -224,37 +226,15 @@ def __call__(self, spectral_info):
         pref = self.update_pref(spectral_info.pref)
         return spectral_info._replace(carriers=carriers, pref=pref)
 
-FiberParams = namedtuple('FiberParams', 'type_variety length loss_coef length_units \
-                                         att_in con_in con_out dispersion gamma')
 
 class Fiber(Node):
     def __init__(self, *args, params=None, **kwargs):
-        if params is None:
+        if not params:
             params = {}
-        if 'con_in' not in params:
-            # if not defined in the network json connector loss in/out
-            # the None value will be updated in network.py[build_network]
-            # with default values from eqpt_config.json[Spans]
-            params['con_in'] = None
-            params['con_out'] = None
-        if 'att_in' not in params:
-            #fixed attenuator for padding
-            params['att_in'] = 0
-
         super().__init__(*args, params=FiberParams(**params), **kwargs)
-        self.type_variety = self.params.type_variety
-        self.length = self.params.length * UNITS[self.params.length_units] # in m
-        self.loss_coef = self.params.loss_coef * 1e-3 # lineic loss dB/m
-        self.lin_loss_coef = self.params.loss_coef / (20 * log10(exp(1)))
-        self.att_in = self.params.att_in
-        self.con_in = self.params.con_in
-        self.con_out = self.params.con_out
-        self.dispersion = self.params.dispersion  # s/m/m
-        self.gamma = self.params.gamma # 1/W/m
-        self.pch_out_db = None
         self.carriers_in = None
         self.carriers_out = None
-        # TODO|jla: discuss factor 2 in the linear lineic attenuation
+        self.nli_solver = NliSolver(self)
```

Comment:
This has been addressed in the recent commit e33cfa3. To clarify, this attribute is currently only used within science_utils.propagate_raman_fiber.

Reply:
If you only use it in the `RamanFiber` class, why do you instantiate `NliSolver` in the `Fiber` class?

Reply:
We do this as we intend to restructure the propagation of the fiber (currently fully defined within element.fiber) to instead use the NLISolver (to avoid again a duplication of the code).